components/collection/activity/Activity.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="is-flex">
+    <SidebarFilter />
+    <div ref="wrapper" class="w-full mt-5">
+      <div v-if="tablet">
+        <div class="columns">
+          <div class="column is-two-thirds">
+            <ActivityChart :events="events" />
+          </div>
+          <div class="column">
+            <OwnerInsights :owners="owners" :flippers="flippers" />
+          </div>
+        </div>
+        <BreadcrumbsFilter />
+      </div>
+      <div v-if="mobile">
+        <div class="is-flex is-flex-direction-column gap">
+          <OwnerInsights :owners="owners" :flippers="flippers" />
+          <div class="max-width">
+            <ActivityChart :events="events" />
+          </div>
+        </div>
+      </div>
+      <hr class="my-5" />
+      <Events :events="sortedEvents" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import ActivityChart from './Chart.vue'
+import OwnerInsights from './OwnerInsights.vue'
+import Events from './events/Events.vue'
+import BreadcrumbsFilter from '@/components/shared/BreadcrumbsFilter.vue'
+import { useCollectionActivity } from '@/components/collection/utils/useCollectionActivity'
+import { Interaction } from '@kodadot1/minimark'
+import { useResizeObserver } from '@vueuse/core'
+import SidebarFilter from '@/components/shared/filters/SidebarFilter.vue'
+const mobileBreakpoint = 800
+const route = useRoute()
+const tablet = ref(true)
+const mobile = ref(false)
+const wrapper = ref<HTMLDivElement | null>(null)
+
+const collectionId = computed(() => route.params.id)
+const { events, flippers, owners, offers } = useCollectionActivity({
+  collectionId: collectionId.value,
+})
+
+const InteractionIncluded = [
+  Interaction.BUY,
+  Interaction.LIST,
+  Interaction.MINTNFT,
+  Interaction.SEND,
+]
+
+const filteredEvents = computed(() =>
+  events.value.filter((event) =>
+    InteractionIncluded.includes(event.interaction as Interaction)
+  )
+)
+const withOffers = computed(() => [...filteredEvents.value, ...offers.value])
+
+const sortedEvents = computed(() =>
+  withOffers.value.sort((a, b) => b.timestamp - a.timestamp)
+)
+
+useResizeObserver(wrapper, (entry) => {
+  if (entry[0].contentRect.width >= mobileBreakpoint) {
+    tablet.value = true
+    mobile.value = false
+  } else {
+    tablet.value = false
+    mobile.value = true
+  }
+})
```

Comment:
pls only keep one state `isMobile` or `isTablet`

Reply:
yep. thank you :)
fixed